LLMonFHIR/FHIR Display/OpenAIChatView.swift

```diff
@@ -87,4 +77,109 @@ struct OpenAIChatView: View {
             }
         }
     }
+    
+    private func processMultipleResourceChat() async throws {
+        let resourcesArray = await fhirStandard.resources
+        var stringResourcesArray = resourcesArray.map { "\($0.displayName) in \($0.resourceType)" }
+        stringResourcesArray.append("N/A")
+        let functionCallOutputArray = try await getFunctionCallOutputArray(stringResourcesArray)
+        
+        if !functionCallOutputArray.contains(where: { $0 == "N/A" }) && !functionCallOutputArray.contains(where: { $0 == " N/A" }) {
+            processFunctionCallOutputArray(functionCallOutputArray: functionCallOutputArray, resourcesArray: resourcesArray)
+        }
+    }
+
+    private func getFunctionCallOutputArray(_ stringResourcesArray: [String]) async throws -> [String] {
+        let openAI = OpenAI(apiToken: openAPIComponent.apiToken ?? "")
+        
+        let functions = [
+            ChatFunctionDeclaration(
+                name: "get_resource_titles",
+                description: String(localized: "MULTIPLE_RESOURCE_FUNCTION_DESCRIPTION"),
+                parameters: JSONSchema(
+                    type: .object,
+                    properties: [
+                        "resources": .init(
+                            type: .string,
+                            description: String(localized: "MULTIPLE_RESOURCE_PARAMETER_DESCRIPTION"),
+                            enumValues: stringResourcesArray
+                        )
+                    ],
+                    required: ["resources"]
+                )
+            )
+        ]
+        
+        let query = ChatQuery(
+            model: "gpt-3.5-turbo-0613",
```

Comment:
We should use the model that is configured in the OpenAI component here, the user can have the choice between different models during the onboarding flow.
We should probably only offer the function calling one here. Not sure if this requires a change in SpeziML. If that is the case, then we should create a nice issue over there to make the choices of models an option that can be configured when creating an OpenAI component.